src/v/cluster/node_status_backend.cc

```diff
@@ -172,6 +174,20 @@ void node_status_backend::tick() {
 }
 
 ss::future<> node_status_backend::collect_and_store_updates() {
+    // reset connections if needed
+    co_await ssx::parallel_transform(
+      _discovered_peers.begin(),
+      _discovered_peers.end(),
+      [this](auto node_and_data) -> ss::future<> {
+          auto& [node, data] = node_and_data;
+          if (data.consecutive_timeouts > consecutive_timeout_reconnect) {
+              _discovered_peers[node].consecutive_timeouts = 0;
+              vlog(clusterlog.info, "Forcing reconnection of peer: {}", node);
```

Comment:
nit: ... due to X consecutive failures (consider adding more context)

also this is going to be chatty with the defaults 20 timeouts and 100ms interval, logged every 2s on every (alive) broker, perhaps debug? (or use a ratelimited logger)